cheatsheets/Microservices_Security_Cheat_Sheet.md

```diff
@@ -2,176 +2,660 @@
 
-- Pushing all authorization decisions to the API gateway can quickly become hard to manage in complex ecosystems with many roles and access control rules.
-- The API gateway may become a single point of decision that may violate the “defense in depth” principle.
-- Operation teams typically own the API gateway, so development teams cannot directly make authorization changes, slowing down velocity due to additional communication and process overhead.
-  
-In most cases, development teams implement authorization in both places – at the edge level at a coarse level of granularity, and at service level. To authenticate an external entity, the edge can use access tokens (referenced token or self-contained token) transmitted via HTTP headers (e.g., “Cookie” or “Authorization”) or use mTLS.
+## Authorization Reference Architecture
 
-## Service-level authorization
+To lay the foundation for the patterns described in this cheat sheet, this section introduces the general building blocks of an authorization system, based on [NIST SP 800-162](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-162.pdf). While that standard focuses on Attribute-Based Access Control (ABAC), the architectural components it defines are relevant to nearly any access control system.
 
-Service-level authorization gives each microservice more control to enforce access control policies.
-For further discussion, we will use terms and definitions according with [NIST SP 800-162](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-162.pdf). The functional components of an access control system can be classified as follows:
+![Authorization Reference Architecture](../assets/Authorization_Reference_Architecture.svg)
 
-- Policy Administration Point (PAP): Provides a user interface for creating, managing, testing, and debugging access control rules.
-- Policy Decision Point (PDP): Computes access decisions by evaluating the applicable access control policy.
-- Policy Enforcement Point (PEP): Enforces policy decisions in response to a request from a subject requesting access to a protected object.
-- Policy Information Point (PIP): Serves as the retrieval source of attributes or the data required for policy evaluation to provide the information needed by the PDP to make decisions.
+These functional components are:
 
-![NIST ABAC framework](../assets/NIST_ABAC.png)
+* **Policy Administration Point (PAP):** Manages policies—offering tools for writing, testing, and updating access control logic.
+* **Policy Decision Point (PDP):** Evaluates policies and computes decisions about access based on the incoming request and relevant attributes.
+* **Policy Enforcement Point (PEP):** Intercepts requests and enforces the access decision provided by the PDP.
+* **Policy Information Point (PIP):** Supplies attribute data or contextual information that the PDP requires to evaluate a policy.
```

Comment:
Recommendation: I'd add a short definition of Subject and Object here, too. 

Maybe also one for the repositories, though they are not necessarily different components depending on the architectural pattern (from "everything in application-db" to the picture shown above.)?

Reply:
Yeah, I was thinking about that as well, but decided to address those elements in the story section.

Repositories are indeed not described explicitly. Because of the name build from "context" and "repository" I thought those elements are self-descriptive enough.

But you're right, I should add a short definition for each to avoid scattering of information throughout the document.